VideoLocker/src/main/java/org/edx/mobile/http/Api.java

```diff
@@ -1223,6 +1221,12 @@ public RegistrationDescription getRegistrationDescription() throws Exception {
     }
 
     public boolean enrollInACourse(String courseId, boolean email_opt_in) throws Exception {
+        // uses auth headers
+        Bundle headers = getAuthHeaders();
+
+        // this API endpoint also requires CSRFToken, so add it
+        headers = setCookieHeaders(headers);
```

Comment:
How much of a pain would it be to change the code so this is set on every authenticated post?